source/clear-linux/guides/maintenance/swupd-mixer-integration.rst

```diff
@@ -0,0 +1,116 @@
+command like so::
+
+  # swupd bundle-add mixer
+
+What is it
+================
+The software updater now has native integration with the mixer tool. One of the
+highest requests for Clear Linux OS for Intel Architecture was the ability for
+users to add their own custom content to their OS. Mixer initially enabled this,
+but it required users to spin their own full OS, and be the OSV for it making
+all the releases and update content.
+
+With this new feature, users can now add custom content ``without`` having to
+compose the entire OS themselves. For example, a user may want a package that
+does not exist in our OS yet. Creating an entire OS just to have a single package
+is doable, but not the correct way to solve the problem. Instead, users can now
+add custom packages and bundles using the :file:`add-pkg.sh` script, and simply
+calling :command:`swupd update && swupd bundle add <new_bundle>` to add just
+their new content. The regular upstream content will still be pulled from the
+normal URL, while local content will be added from a local update store. *Note*
+that while this feature is intended to enable additive, unique content, it is
+still possible to add non-unique packages, i.e replacing something already provided
+by upstream with a local version. However that is not guaranteed to always work, thus
+the updater will warn the user if conflicts are found before creating the local content.
+
+How it works
+============
+The software updater now checks to see if there is any content in :file:`/usr/share/mix`.
+If content does exist, then it knows user content must be incorporated into the updater
+commands. Internally it does all the same operations to retrieve the upstream manifests
+and content, and will also do the same for the local content. Upstream Manifest.MoMs
+are still signature verified, so users are guaranteed all content not provided by their
+local system is secure and correct.
+
+When a user calls :file:`add-pkg.sh`, the script performs all the necessary mixer
+commands and bookeeping to create a minimal mix. This mix contains only the new user
+bundle, and os-core (to keep track of the versioning and timestamp changes). To keep
+swupd from becoming too complex and multiplexing seperate Manifest.MoMs, the script also
+merges the upstream Manifest.MoM (once it is openssl verified), with the local Manifest.MoM.
+The result is a merged Manifest.MoM that contains any updates from upstream, with the
+local updates concantenated to it. The new merged manifest is then signed with the user's
+certificate, which swupd will use to verify the manifest is the one it should be.
+
+Consuming only one Manifest.MoM relieves some burden from the updater, which now can
+simply read it, and understand if a bundle originates from upstream, or from local. If
+a maniest is tagged as local in the Manifest.MoM, the updater will load that manifest
+from disc instead of going over the network, and all files from it will be marked as
```

Comment:
Had to look this up, but `s/disc/disk/` for non-optical drives.

> Disk is the standard spelling for computer-related terms such as hard disk and floppy disk. Disc is the standard spelling for phonograph records, albums (in the figurative sense—a group of songs presented in sequence), and components of plows and brake systems.

http://grammarist.com/spelling/disc-disk/